conan/api/subapi/cache.py

```diff
@@ -182,14 +182,10 @@ def restore(self, path):
 
         cache = PkgCache(self.conan_api.cache_folder, self.conan_api.config.global_conf)
         cache_folder = cache.store  # Note, this is not the home, but the actual package cache
-
-        with open(path, mode='rb') as file_handler:
-            the_tar = tarfile.open(fileobj=file_handler)
-            fileobj = the_tar.extractfile("pkglist.json")
-            pkglist = fileobj.read()
-            the_tar.extraction_filter = (lambda member, _: member)  # fully_trusted (Py 3.14)
-            the_tar.extractall(path=cache_folder)
-            the_tar.close()
+        tar_extract(path, cache_folder, cache_folder=self.conan_api.cache_folder)
```

Comment:
Try to use here direclty the idiom ``if plugin -> use plugin``, instead of abstracting away the ``tar_extract()``